app/models/user.rb

```diff
@@ -0,0 +1,39 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :name, String
+  property :email, String
+  property :username, String
+  property :password_digest, Text
+
+  has n, :peeps, :through => Resource
+
+  # Password encryption
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(@password)
+  end
+
+  # Authentication
+  def self.authenticate(email, password)
+    user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      # return this user
+      user
+    else
+      nil
+    end
+  end
+
+  # Validations
+  validates_confirmation_of :password
+  validates_format_of :email, as: :email_address, message: 'Please enter a valid email address mate'
```

Comment:
Metrics/LineLength: Line is too long. [100/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)